s3torchconnectorclient/rust/src/mountpoint_s3_client.rs

```diff
@@ -42,26 +44,33 @@ pub struct MountpointS3Client {
     no_sign_request: bool,
     #[pyo3(get)]
     user_agent_prefix: String,
+    #[pyo3(get)]
+    endpoint: String,
 
     owner_pid: Pid,
 }
 
 #[pymethods]
 impl MountpointS3Client {
     #[new]
-    #[pyo3(signature = (region, user_agent_prefix="".to_string(), throughput_target_gbps=10.0, part_size=8*1024*1024, profile=None, no_sign_request=false))]
+    #[pyo3(signature = (region, user_agent_prefix="".to_string(), throughput_target_gbps=10.0, part_size=8*1024*1024, profile=None, no_sign_request=false, endpoint="".to_string()))]
     pub fn new_s3_client(
         region: String,
         user_agent_prefix: String,
         throughput_target_gbps: f64,
         part_size: usize,
         profile: Option<String>,
         no_sign_request: bool,
+        endpoint: String,
     ) -> PyResult<Self> {
         // TODO: Mountpoint has logic for guessing based on instance type. It may be worth having
         // similar logic if we want to exceed 10Gbps reading for larger instances
 
-        let endpoint_config = EndpointConfig::new(&region);
+        let endpoint_config = if endpoint.is_empty() {
+            EndpointConfig::new(&region)
+        } else {
+            EndpointConfig::new(&region).endpoint(Uri::new_from_str(&Allocator::default(), endpoint.clone()).unwrap())
```

Comment:
Can we see if we need to make changes here if an invalid URI is passed? (And associated unit tests).

Reply:
Changed the default value of endpoint parameter to None instead of empty string. If an invalid URI is passed the underneath code in moutpoits3 will report error